Update blip model card (#38513)
huggingface/transformers: docs/source/en/model_doc/blip.md

-->

<div style="float: right;">
    <div class="flex flex-wrap space-x-1">
        <img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
        <img alt="TensorFlow" src="https://img.shields.io/badge/TensorFlow-FF6F00?style=flat&logo=tensorflow&logoColor=white">
    </div>
</div>

# BLIP

> [BLIP](https://huggingface.co/papers/2201.12086) (Bootstrapped Language-Image Pretraining) is a multimodal model that can understand and generate images and text together, and it is distinctive in that one model can handle various vision-language tasks such as image captioning, image-text matching, and VQA all together. Unlike existing VLP models that were specialized in only one of understanding or generation, BLIP is designed to be flexibly transferred to both domains. In particular, without using noisy image-text pairs collected from the web as they are, it introduces a ‘bootstrapping caption’ technique that generates sentences with its own captioner and goes through a filtering process to increase learning quality, thereby securing cleaner and more meaningful data. As a result, BLIP achieves high performance and excellent generalization despite little artificial manual labor, and can be effectively utilized for various multimodal tasks.

Reviewer: ```suggestion
[BLIP](https://huggingface.co/papers/2201.12086) (Bootstrapped Language-Image Pretraining) is a vision-language pretraining (VLP) framework designed for *both* understanding and generation tasks. Most existing pretrained models are only good at one or the other. It uses a captioner to generate captions and a filter to remove the noisy captions. This increases training data quality and more effectively uses the messy web data.
```

You can find all the original BLIP checkpoints under the [BLIP](https://huggingface.co/collections/Salesforce/blip-models-65242f40f1491fbf6a9e9472) collection.

> [!TIP]

> > Click on the BLIP models in the right sidebar for more examples of how to apply BLIP to different tasks.

Reviewer: ```suggestion
> Click on the BLIP models in the right sidebar for more examples of how to apply BLIP to different vision language tasks.
```

> The example below demonstrates how to generate text based on an image with [`Pipeline`] or the [`AutoModel`] class.

Reviewer: ```suggestion
The example below demonstrates how to visual question answering with [`Pipeline`] or the [`AutoModel`] class.
```

<hfoptions id="usage">
<hfoption id="Pipeline">

```python

> import torch

Reviewer: Lets use a more interesting image!

```py
import torch
from transformers import pipeline

pipeline = pipeline(
    task="visual-question-answering",
    model="Salesforce/blip-vqa-base",
    torch_dtype=torch.float16,
    device=0
)
url = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg"
pipeline(question="What is the weather in this image?", image=url)
```

from transformers import pipeline

pipeline = pipeline(
    task="visual-question-answering",
    model="Salesforce/blip-vqa-base",
    torch_dtype=torch.float16,
    device=0
)
url = "http://images.cocodataset.org/val2017/000000039769.jpg"
print(pipeline(question="What is cat doing?", image=url))
```

</hfoption>

> <hfoption id="AutoModel">

Reviewer: Same with the `AutoModel` example.

```py
import requests
import torch
from PIL import Image
from transformers import AutoProcessor, AutoModelForVisualQuestionAnswering

processor = AutoProcessor.from_pretrained("Salesforce/blip-vqa-base")
model = AutoModelForVisualQuestionAnswering.from_pretrained(
    "Salesforce/blip-vqa-base", 
    torch_dtype=torch.float16,
    device_map="auto"
)

url = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg"
image = Image.open(requests.get(url, stream=True).raw)

question = "What is the weather in this image?"
inputs = processor(images=image, text=question, return_tensors="pt").to("cuda", torch.float16)

output = model.generate(**inputs)
processor.batch_decode(output, skip_special_tokens=True)[0]
```

```python
import requests
import torch
from PIL import Image
from transformers import AutoProcessor, AutoModelForVisualQuestionAnswering

processor = AutoProcessor.from_pretrained("Salesforce/blip-vqa-base")
model = AutoModelForVisualQuestionAnswering.from_pretrained(
    "Salesforce/blip-vqa-base", 
    torch_dtype=torch.float16,
    device_map="auto"
)

url = "http://images.cocodataset.org/val2017/000000039769.jpg"
image = Image.open(requests.get(url, stream=True).raw)

question = "What is cat doing?"
inputs = processor(images=image, text=question, return_tensors="pt").to("cuda", torch.float16)

output = model.generate(**inputs)
print(processor.batch_decode(output, skip_special_tokens=True)[0])
```

</hfoption>

</hfoptions>

![BLIP.gif](https://cdn-uploads.huggingface.co/production/uploads/1670928184033-62441d1d9fdefb55a0b7d12c.gif)

> ## Notes
> 
> - This model was contributed by [ybelkada](https://huggingface.co/ybelkada).
> - The original code can be found [here](https://github.com/salesforce/BLIP).

Reviewer: ```suggestion
```

## Resources

> 

Reviewer: For the notebook below, lets use this language:

```md
Refer to this [notebook](https://github.com/huggingface/notebooks/blob/main/examples/image_captioning_blip.ipynb) to learn how to fine-tune BLIP for image captioning on a custom dataset.
```
